test/lib/jdk/test/lib/process/ProcessTools.java

```diff
@@ -181,6 +241,12 @@ public static Process startProcess(String name,
 
         stdout.addPump(new LineForwarder(name, System.out));
         stderr.addPump(new LineForwarder(name, System.err));
+        BufferInputStream out = new BufferInputStream(p);
+        BufferInputStream err = new BufferInputStream(p);
+
+        stdout.addOutputStream(out.getOutputStream());
+        stderr.addOutputStream(err.getOutputStream());
+
```

Comment:
Overall this looks good, although I'm a bit unclear on how some of the underpinnings work, allowing the output to appear in these output streams, and also in the LineForwarder output (above), and for that matter, in the lineConsumer output (below). I guess there is some multiplexing of the output that I just don't grasp, but appears to be something that already worked.

Reply:
StreamPumper read process stdout, stderr streams and write read data to registered streams. So it works as multiplexer.